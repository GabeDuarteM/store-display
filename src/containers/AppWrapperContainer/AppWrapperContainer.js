import React from 'react'

import PropTypes from 'prop-types'

import configureStore from '../../store'
import AppWrapper from '../../components/AppWrapper'

const store = configureStore()

const AppWrapperContainer = ({ children }) => (
  <AppWrapper store={store}>{children}</AppWrapper>
)

export const propTypes = {
  /** The children that is supposed to be rendered */
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
}

AppWrapperContainer.propTypes = propTypes

export default AppWrapperContainer
