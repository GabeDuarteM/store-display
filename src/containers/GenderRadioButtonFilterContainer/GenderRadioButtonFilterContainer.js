import React from 'react'

import { connect } from 'react-redux'
import PropTypes from 'prop-types'

import RadioButtonFilter from '../../components/RadioButtonFilter'
import { filterChangeGender as filterChangeGenderAction } from '../../store/reducers/filter/filterActions'
import { productRefreshList } from '../../store/reducers/product/productActions'

export const GenderRadioButtonFilterContainer = ({
  activeOption,
  filterChangeGender,
}) => (
  <RadioButtonFilter
    options={[{ id: 'woman', text: 'Woman' }, { id: 'man', text: 'Man' }]}
    activeOption={activeOption}
    onChange={filterChangeGender}
  />
)

const propTypes = {
  /** @ignore */
  activeOption: PropTypes.oneOf(['man', 'woman']).isRequired,
  /** @ignore */
  filterChangeGender: PropTypes.func.isRequired,
}

GenderRadioButtonFilterContainer.propTypes = propTypes

const mapStateToProps = state => ({
  activeOption: state.filter.gender,
})

const mapDispatchToProps = dispatch => ({
  filterChangeGender: gender => {
    dispatch(filterChangeGenderAction(gender))
    dispatch(productRefreshList())
    window.scrollTo(0, 0)
  },
})

export default connect(mapStateToProps, mapDispatchToProps)(
  GenderRadioButtonFilterContainer,
)
