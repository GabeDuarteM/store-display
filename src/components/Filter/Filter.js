import React from 'react'
import PropTypes from 'prop-types'
import injectSheet from 'react-jss'
import cn from 'classnames'

import GenderRadioButtonFilterContainer from '../../containers/GenderRadioButtonFilterContainer'
import SortDropdownFilterContainer from '../../containers/SortDropdownFilterContainer'

export const styles = {
  filter: {
    backgroundColor: 'white',
    paddingTop: 20,
    paddingBottom: 25,
    paddingLeft: 62,
    paddingRight: 62,
    display: 'flex',
    justifyContent: 'space-between',
  },
}

const Filter = ({ classes, className }) => (
  <div className={cn(classes.filter, className)}>
    <GenderRadioButtonFilterContainer />
    <SortDropdownFilterContainer />
  </div>
)

export const propTypes = {
  /** @ignore */
  classes: PropTypes.object.isRequired,
  /** Apply a classname to the root component */
  className: PropTypes.string,
}

export const defaultProps = {
  className: null,
}

Filter.propTypes = propTypes
Filter.defaultProps = defaultProps

export default injectSheet(styles)(Filter)
