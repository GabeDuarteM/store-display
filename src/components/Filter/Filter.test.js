import React from 'react'

import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'

import Filter, { styles, propTypes, defaultProps } from './Filter'
import GenderRadioButtonFilterContainer from '../../containers/GenderRadioButtonFilterContainer'
import AppWrapperContainer from '../../containers/AppWrapperContainer'

const createWrapperShallow = () => shallow(<Filter />)

const createWrapperMount = () =>
  mount(
    <AppWrapperContainer>
      <Filter />
    </AppWrapperContainer>,
  )

describe('COMPONENT: <Filter />', () => {
  it('should render without crashing with default props', () => {
    createWrapperMount()
  })

  it("should match component's snapshot", () => {
    const wrapper = createWrapperShallow()
    expect(toJson(wrapper.dive())).toMatchSnapshot()
  })

  it("should match component's style snapshot", () => {
    expect(styles).toMatchSnapshot()
  })

  it("should match component's propTypes snapshot", () => {
    expect(propTypes).toMatchSnapshot()
  })

  it("should match component's defaultProps snapshot", () => {
    expect(defaultProps).toMatchSnapshot()
  })

  it('should render a GenderRadioButtonFilterContainer', () => {
    const wrapper = createWrapperShallow()
    expect(
      wrapper
        .dive()
        .find(GenderRadioButtonFilterContainer)
        .exists(),
    ).toBeTruthy()
  })
})
