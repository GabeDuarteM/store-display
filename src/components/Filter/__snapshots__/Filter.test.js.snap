// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT: <Filter /> should match component's defaultProps snapshot 1`] = `
Object {
  "className": null,
}
`;

exports[`COMPONENT: <Filter /> should match component's propTypes snapshot 1`] = `
Object {
  "className": [Function],
  "classes": [Function],
}
`;

exports[`COMPONENT: <Filter /> should match component's snapshot 1`] = `
<div
  className="Filter-filter"
>
  <Connect(GenderRadioButtonFilterContainer) />
  <Connect(SortDropdownFilterContainer) />
</div>
`;

exports[`COMPONENT: <Filter /> should match component's style snapshot 1`] = `
Object {
  "filter": Object {
    "backgroundColor": "white",
    "display": "flex",
    "justifyContent": "space-between",
    "paddingBottom": 25,
    "paddingLeft": 62,
    "paddingRight": 62,
    "paddingTop": 20,
  },
}
`;
